fix: nuxt-icon and add tree shaking to icons (#97)
cpsoinos/nuxt-svgo: README.md

Then, in any `.vue` file, import your asset and use it as a component:

```vue
<template>
  <div>
    <IconHome class="w-5 h-5" />
  </div>
</template>

<script setup lang="ts">
import IconHome from '~/assets/icon-home.svg'
</script>
```

Or, if you use **vite**, in any `.vue` file, simply use your icon's name with `svgo` prefix as component name:

```vue
<template>
  <div>
    <SvgoHome class="w-5 h-5" />
    <!-- Or -->
    <svgo-home class="w-5 h-5" />
  </div>
</template>
```

It automatically imports your icons from `assets/icons/` folder by default. you can configure this by passing `autoImportPath` in your config:

```typescript
// nuxt.config.ts
import { defineNuxtConfig } from 'nuxt'

export default defineNuxtConfig({
  modules: ['nuxt-svgo'],
  svgo: {
    autoImportPath: './assets/other-icons/'
  }
})
```

If you want to use auto import but you don't want to use the `nuxt-icon` component (used by default), You can do so by using `simpleAutoImport: true`:

```typescript
// nuxt.config.ts
import { defineNuxtConfig } from 'nuxt'

export default defineNuxtConfig({
  modules: ['nuxt-svgo'],
  svgo: {
    simpleAutoImport: true
  }
})
```

### Subfolders

> The icons's component name will follow nuxt's component prefix convention. Therefore, if prefix is turned on for your components, the component name for example `assets/icons/admin/badge.svg` will be `svgo-admin-badge`:

Reviewer: ```suggestion
The icons's component name will follow Nuxt's component prefix convention. Therefore, if prefix is turned on for your components, the component name for `assets/icons/admin/badge.svg`, for example, will be `svgo-admin-badge`:
```

## `nuxt-icon` component

> [originally copied over from nuxt-icons module](https://github.com/gitFoxCode/nuxt-icons/blob/89e53649e5868c31fc97869918ede96504ae1a04/src/runtime/components/nuxt-icon.vue), but later heavily modified to support tree shaking and SSR. This is not intended to be used directly. However you can import your Icons directly and pass it to the component using `icon` property

Reviewer: ```suggestion
Originally copied over from the [nuxt-icons module](https://github.com/gitFoxCode/nuxt-icons/blob/89e53649e5868c31fc97869918ede96504ae1a04/src/runtime/components/nuxt-icon.vue), but later heavily modified to support tree shaking and SSR. This is not intended to be used directly. However, you can import your icons directly and pass them to the component using the `icon` prop.
```
